### ISSUE TITLE
Output everything as json

### ISSUE THREAD

**Issue by Wulfheart** [code-lts/doctum#29]
Instead of html output I'd like to generate a json for each class and the navigation. Do you have any idea how I could achieve this? 

Thank you in advance for your help.

**Comment by williamdes** [comment]
Hi !
I think you can acheive this by using a theme
Here is an example to build markdown output: https://github.com/code-lts/Laravel-FCM/tree/main/scripts

Twig has json_encode functions, let me know if you need some help

**Comment by williamdes** [comment]
Hi !
I think you can acheive this by using a theme
Here is an example to build markdown output: https://github.com/code-lts/Laravel-FCM/tree/main/scripts

Twig has json_encode functions, let me know if you need some help

**Comment by Wulfheart** [comment]
Thanks for your quick answer.
But you don't know of an already premade template, don't you?

**Comment by williamdes** [comment]
> Thanks for your quick answer.
> But you don't know of an already premade template, don't you?

No, I only know an implementation of ascii docs but nothing with JSON

Why do you need the JSON format by the way ?

**Comment by Wulfheart** [comment]
I'd like to make the docs available in a machine readable format.

**Comment by williamdes** [comment]
> I'd like to make the docs available in a machine readable format.

Okay, maybe the best would be to create a command like the "parse" one to ouput the docs to JSON

That would help, that said some JSON files are available in the cache folder after a parse operation.

**Comment by williamdes** [comment]
> I'd like to make the docs available in a machine readable format.

Do you have a specific idea in mind, maybe we can workout a solution ?

**Comment by Wulfheart** [comment]
I'm planning to store it in a database or as json to do analysis on it.

**Comment by williamdes** [comment]
> I'm planning to store it in a database or as json so I can render it dynamically, not generated in advance.

Oh this is the behaviour of `parse` command, and then you can call `render` when you want

**Comment by williamdes** [comment]
Hi @Wulfheart 
The new search index generated since bc0be8a988e2766a452e9338657ff47de68462d8 will probably be usefull if you want to hack into the tool, that said the file definition can change anytime.
